### PR TITLE
ci: run schedule on upstream only and upgrade actions version

### PR DESCRIPTION
**Pull request by tisonkun**
cc @ZonaHex @alili

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -71,9 +71,10 @@ jobs:
     needs: [build, release]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3.1.0
+      - uses: actions/checkout@v4
         with:
           repository: 'GreptimeTeam/greptimedb'
+          token: ${{ secrets.PR_ACTION }}
```
Comment (sunng87) [inline .github/workflows/release.yml:77]: Is this token required here?

Comment (sunng87) [reply .github/workflows/release.yml:77]: It should be the token of line 83. I doubt if the token has expired.

Comment (tisonkun) [reply .github/workflows/release.yml:77]: So I'd revert this change line and you check the secrets or replace it with a new generated PAT?

Comment (sunng87) [reply .github/workflows/release.yml:77]: I need to confirm with @ZonaHex if she has the token

Comment (tisonkun) [reply .github/workflows/release.yml:77]: Well. Let me revert this change line and we can review this topic later. We can still merge the other changes >_<

**Review by ZonaHex — APPROVED** [review]
LGTM

**Pull request merged**